pkg/executor/executor.go

```diff
@@ -574,6 +590,40 @@ func (ex *executor) doHTTPRequestWithContext(operation diagnosisv1.Operation, da
 	return true, result, nil
 }
 
+// runScriptWithContext runs a script with the arguments provided by context.
+// It returns a bool, a map and an error as results.
+func (ex *executor) runScriptWithContext(operation diagnosisv1.Operation, data map[string]string) (bool, map[string]string, error) {
+	if operation.Spec.Processor.ScriptRunner == nil {
+		return false, nil, fmt.Errorf("script runner not specified")
+	}
+
+	// Generate all argument according to script runner definition and execute the script with timeout.
+	var args []string
+	for _, key := range operation.Spec.Processor.ScriptRunner.ArgKeys {
+		if value, ok := data[key]; ok {
+			args = append(args, value)
+		}
+	}
+	scriptFilePath := filepath.Join(ex.dataRoot, controllers.ScriptSubDirectory, operation.Name)
+	command := append([]string{"/bin/sh", scriptFilePath}, args...)
+	output, err := util.BlockingRunCommandWithTimeout(command, *operation.Spec.Processor.TimeoutSeconds)
```

Comment:
Will it be possible that the script need some environment variables from diagnosis parameters or results of front operation？

Reply:
A script might depends on environment variables. However, script runner executes a script on a node, which is different from the scenario of running a program in a container. The context propagation model is sufficient for a user to set environment variables in another script by taking arguments.